Source/SwiftLintFramework/Rules/Style/ProtocolPropertyAccessorsOrderRule.swift

```diff
@@ -25,19 +24,75 @@ public struct ProtocolPropertyAccessorsOrderRule: ConfigurationProviderRule, Sub
         ]
     )
 
-    public func validate(file: SwiftLintFile) -> [StyleViolation] {
-        return violationRanges(in: file).map {
-            StyleViolation(ruleDescription: Self.description,
-                           severity: configuration.severity,
-                           location: Location(file: file, characterOffset: $0.location))
+    public func makeVisitor(file: SwiftLintFile) -> ViolationsSyntaxVisitor? {
+        Visitor()
+    }
+
+    public func makeRewriter(file: SwiftLintFile) -> ViolationsSyntaxRewriter? {
+        file.locationConverter.map { locationConverter in
+            Rewriter(
+                locationConverter: locationConverter,
+                disabledRegions: disabledRegions(file: file)
+            )
         }
     }
+}
+
+private extension ProtocolPropertyAccessorsOrderRule {
+    final class Visitor: SyntaxVisitor, ViolationsSyntaxVisitor {
+        private(set) var violationPositions: [AbsolutePosition] = []
+
+        override func visitPost(_ node: AccessorBlockSyntax) {
+            guard node.hasViolation else {
+                return
+            }
 
-    public func violationRanges(in file: SwiftLintFile) -> [NSRange] {
-        return file.match(pattern: "\\bset\\s*get\\b", with: [.keyword, .keyword])
+            violationPositions.append(node.accessors.positionAfterSkippingLeadingTrivia)
+        }
     }
 
-    public func substitution(for violationRange: NSRange, in file: SwiftLintFile) -> (NSRange, String)? {
-        return (violationRange, "get set")
+    private final class Rewriter: SyntaxRewriter, ViolationsSyntaxRewriter {
+        private(set) var correctionPositions: [AbsolutePosition] = []
+        let locationConverter: SourceLocationConverter
+        let disabledRegions: [SourceRange]
+
+        init(locationConverter: SourceLocationConverter, disabledRegions: [SourceRange]) {
+            self.locationConverter = locationConverter
+            self.disabledRegions = disabledRegions
+        }
+
+        override func visit(_ node: AccessorBlockSyntax) -> Syntax {
+            guard node.hasViolation else {
+                return super.visit(node)
+            }
+
+            let isInDisabledRegion = disabledRegions.contains { region in
+                region.contains(node.positionAfterSkippingLeadingTrivia, locationConverter: locationConverter)
+            }
+
+            guard !isInDisabledRegion else {
+                return super.visit(node)
+            }
+
+            correctionPositions.append(node.accessors.positionAfterSkippingLeadingTrivia)
+
+            let reversedAccessors = SyntaxFactory.makeAccessorList(Array(node.accessors.reversed()))
+            return super.visit(
+                node.withAccessors(reversedAccessors)
+            )
+        }
+    }
+}
+
+private extension AccessorBlockSyntax {
+    var hasViolation: Bool {
+        guard accessors.count == 2,
+              accessors.allSatisfy({ $0.body == nil }),
```

Comment:
What is the body of an accessor? 🤔

Reply:
an `AcessorDeclSyntax` is also used for computed properties, so having a body means it's not from a protocol. For example:

```swift
struct A {
  var b: Int {
    get { return 1 }
  }
}
```

Reply:
I see. As only protocols are relevant for this rule, traversing the AST could be stopped as soon as a class, struct, enum or extension is encountered. This is done for example in `EmptyXCTestMethodRule`. This might also accelerate the rule a bit since processing can be stopped quite early. In addition, the check for existing bodies could be skipped.